Library/Homebrew/cask/lib/hbc/system_command.rb

```diff
@@ -56,14 +56,25 @@ def command
 
     private
 
-    attr_reader :executable, :args, :input, :options, :processed_output, :processed_status, :path
+    attr_reader :executable, :args, :input, :options, :processed_output, :processed_status, :env
 
     attr_predicate :sudo?, :print_stdout?, :print_stderr?, :must_succeed?
 
     def sudo_prefix
       return [] unless sudo?
       askpass_flags = ENV.key?("SUDO_ASKPASS") ? ["-A"] : []
-      ["/usr/bin/sudo", *askpass_flags, "-E", "--"]
+      prefix = ["/usr/bin/sudo", *askpass_flags, "-E"]
+
+      env.each do |name, value|
+        name.scan(/^$|^\d|\W/) do
+          ofail "Invalid variable name: '#{name}'"
```

Comment:
Maybe just `raise` instead of `ofail` here. Also not sure if this level of error checking is necessary at all, since this will probably be used in very specific circumstances.

Reply:
I’ve put the checks there specifically to guard against possible [confused deputy attacks](https://en.wikipedia.org/wiki/Confused_deputy_problem), where a malicious cask author may abuse `system_command` to gain root.

For example, let’s suppose some attacker has already managed to plant a malicious `/usr/local/bin/​` payload (note the last character being a `\u200b ZERO WIDTH SPACE`), and is now looking into gaining root. Without additional checks in place, the attacker may find a way to somehow put the `​` character (`\u200b`) inside the hash, and then submit a cask containing `system_command '/usr/bin/some_harmless_thing'` followed by `env` and `sudo: true`. This may then give them root, and maintainers may not notice in time until the damage is done.

I admit my example may be a poor one but it’s just to illustrate my point. There are likely other, more realistic attack vectors, of which we may not be aware right now.

**tl;dr** The checks make sense for security reasons because defense in depth, and because `sudo` is a beast.

I’m fine with `raise` instead of `ofail`, though.

Reply:
Maybe move this into `initialize` and raise an `ArgumentError`.